Source/Particles/Collision/BinaryCollision/DSMC/CollisionFilterFunc.H

```diff
@@ -0,0 +1,228 @@
+/* Copyright 2023 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald (TAE Technologies), Neil Zaim
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef COLLISION_FILTER_FUNC_H_
+#define COLLISION_FILTER_FUNC_H_
+
+#include "Particles/Collision/BinaryCollision/BinaryCollisionUtils.H"
+#include "Particles/Collision/ScatteringProcess.H"
+
+#include <AMReX_Random.H>
+
+/**
+ * \brief This function determines whether a collision occurs for a given
+ * pair of particles.
+ *
+ * @param[in] u1x,u1y,u1z momenta of the first colliding particle
+ * @param[in] u2x,u2y,u2z momenta of the second colliding particle
+ * @param[in] m1,m2 masses
+ * @param[in] w1,w2 effective weight of the colliding particles
+ * @param[in] dt is the time step length between two collision calls.
+ * @param[in] dV is the volume of the corresponding cell.
+ * @param[in] pair_index is the index of the colliding pair
+ * @param[out] p_mask is a mask that will be set to true if fusion occurs for that pair
+ * @param[out] p_pair_reaction_weight stores the weight of the product particles
+ * @param[in] engine the random engine.
+ */
+template <typename index_type>
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void CollisionPairFilter (const amrex::ParticleReal& u1x, const amrex::ParticleReal& u1y,
+                          const amrex::ParticleReal& u1z, const amrex::ParticleReal& u2x,
+                          const amrex::ParticleReal& u2y, const amrex::ParticleReal& u2z,
+                          const amrex::ParticleReal& m1, const amrex::ParticleReal& m2,
+                          amrex::ParticleReal w1, amrex::ParticleReal w2,
+                          const amrex::Real& dt, const amrex::ParticleReal& dV, const int& pair_index,
+                          index_type* AMREX_RESTRICT p_mask,
+                          amrex::ParticleReal* AMREX_RESTRICT p_pair_reaction_weight,
+                          const int& multiplier_ratio,
+                          int const process_count,
+                          ScatteringProcess::Executor* scattering_processes,
+                          const amrex::RandomEngine& engine)
+{
+    using namespace amrex::literals;
+
+    amrex::ParticleReal E_coll, v_coll, lab_to_COM_factor;
+
+    const amrex::ParticleReal w_min = amrex::min(w1, w2);
+    const amrex::ParticleReal w_max = amrex::max(w1, w2);
+
+    BinaryCollisionUtils::get_collision_parameters(
+        u1x, u1y, u1z, u2x, u2y, u2z, m1, m2,
+        E_coll, v_coll, lab_to_COM_factor);
+
+    // convert E_coll to eV
+    E_coll /= PhysConst::q_e;
+
+    amrex::ParticleReal sigma_tot = 0._prt;
+    for (int ii = 0; ii < process_count; ii++) {
+        auto const& scattering_process = *(scattering_processes + ii);
+        sigma_tot += scattering_process.getCrossSection(E_coll);
+    }
+
+    // calculate total collision probability
+    amrex::ParticleReal exponent = (
+        lab_to_COM_factor * multiplier_ratio * w_max
```

Comment:
It's a good point, but I'm not sure it can be applied in quite the same way since in the MCC routine we pre-calculate a maximum collision frequency (which does not take into account any specific colliding pair for which we would know `lab_to_COM_factor` - see https://github.com/ECP-WarpX/WarpX/blob/e460e608a303b642023b2c8418bf5ae2ec64a5f5/Source/Particles/Collision/BackgroundMCC/BackgroundMCCCollision.cpp#L232). The individual collisions are then evaluated based on the ratio of their calculated collision frequency to that maximum collision frequency. In this approach would both the individual and maximum collision frequencies be scaled by the `lab_to_COM_factor`, which would effectively cancel it out?